### DISCUSSION TITLE
Enums with data

### DISCUSSION THREAD

**Discussion by jepcd** [prisma/prisma#16438] [Q&A] [answered]
Hi, I was wondering how I would represent enums with data in my schema

here is an example of what I'm trying to do in rust
```rs
struct Store {
  name: String,
  location: String,
  products: Vec<Product>
}

enum Product {
  Shirt(Shirt),
  Wallet(Wallet)
}

struct Shirt {
  size: String
}

struct Wallet {
  material: String
}
```

**Comment by pantharshit00** [comment, marked as answer]
Hello,

The schema doesn't support polymorphic associations yet so you will need to do something like this. You can track its progress here: https://github.com/prisma/prisma/issues/1644

So your types will convert into something like this:
```prisma
model Store {
  id       String    @id @default(cuid())
  name     String
  location String
  products Product[]
}

model Shirt {
  id      String    @id @default(cuid())
  size    String
  products Product[]
}

model Wallet {
  id        String    @id @default(cuid())
  material  String
  products  Product[]
}

enum ProductType {
  SHIRT
  WALLET
}

model Product {
  id       String      @id @default(cuid())
  type     ProductType
  shirt    Shirt?      @relation(fields: [shirtId], references: [id])
  wallet   Wallet?     @relation(fields: [walletId], references: [id])
  store    Store?      @relation(fields: [storeId], references: [id])
  storeId  String?
  shirtId  String?
  walletId String?
}
```


But honestly in your case I would actually not create tables for each product type and would rather have optional fields in the `Product` table along with enum which will distinguish between product types and you will fill that table accordingly:

```prisma
model Product {
  id       String      @id @default(cuid())
  type     ProductType
  material String?
  size  String?

  store    Store?      @relation(fields: [storeId], references: [id])
  storeId  String?
}
```

**Reply by jepcd** [reply]
That works, but is not feasible when you a significant number of products, each with their own properties.

The only way I've found currently is this
```prisma
model Product {
    id String @id @default(cuid())
    type ProductType
    data Json

    storeId String
    stores Store[]
}
```
and then managing types myself. am I missing something else?

**Reply by pantharshit00** [reply]
Yeah, currently this would be the best solution. You can follow this issue for any updates on Polymorphic Association support in the schema: https://github.com/prisma/prisma/issues/1644

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
